conferences/checkout.go

```diff
@@ -2,33 +2,51 @@ package conferences
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
+	"io/ioutil"
+	"net/http"
 
+	"github.com/stripe/stripe-go/webhook"
+
+	"encore.dev/beta/auth"
+	"encore.dev/storage/sqldb"
 	"github.com/stripe/stripe-go"
 	"github.com/stripe/stripe-go/checkout/session"
 )
 
 // CheckoutParams is the payload of checkout which contains the information about the
 // slot claim purchase about to happen.
 type CheckoutParams struct {
+	ConferenceID uint64
+	VoucherID    string
 	SlotsToClaim []uint64
 }
 
 // CheckoutResponse returns information about the outcome of the checkout initiation
 // process.
 type CheckoutResponse struct {
 	StripeSessionID       string
+	ClaimPayment          *ClaimPayment
 	NoStockOfSlot         *ConferenceSlot
 	MissingSlotDependency *ConferenceSlot
 }
 
+type secrets struct {
+	StripeKey      string
+	EndpointSecret string
+}
+
 // Checkout initiates the checkout process for purchasing slot attendance in a conference.
 func Checkout(ctx context.Context, params *CheckoutParams) (*CheckoutResponse, error) {
-	// Check that we have capacity remaining in the slot and that customer is puchasing
+	usr, ok := auth.Data().(*User)
+	if !ok {
+		return nil, fmt.Errorf("unable to use %T %v as a user", usr, usr)
+	}
+	// Check that we have capacity remaining in the slot and that customer is purchasing
 	// all dependencies.
-	attendeeEmail := "" // FIXME: Find this from user in request.
-	if err := ensureSlotsCanBeClaimed(ctx, params.SlotsToClaim, attendeeEmail); err != nil {
+	if err := ensureSlotsCanBeClaimed(ctx, params.SlotsToClaim, usr.Email); err != nil {
```

Comment:
I believe this "err" referred to "ok" in 43. Let's also make sure that ensureSlotsCanBeClaimed will be created somewhere later

Reply:
not sure what you mean by the `ensureSlotsCanBeClaimed`  comment, it is in https://github.com/gopheracademy/showrunner/pull/24/files#diff-77455d680090041acb94167f9bccc2dfe6d61d50325e20dbdf08693d2caf4cb6R184